tests/Features/ReturnTypes/Helpers/AppExtension.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Features\ReturnTypes\Helpers;
+
+use Illuminate\Foundation\Application;
+use NunoMaduro\Larastan\ApplicationResolver;
+
+class AppExtension
+{
+    public function testObjectType(): ApplicationResolver
+    {
+        return app(ApplicationResolver::class);
+    }
+
+    public function testMixedTypeNoArgument(): Application
+    {
+        return app();
+    }
+
```

Comment:
Maybe we can add another test case here. For example, checking `app('auth')` returns `AuthManager`